src/nativeRenderManager.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Script to handle firing impression and click trackers from native teamplates
+ */
+import { parseUrl, triggerPixel, transformAuctionTargetingData } from './utils';
+import { newNativeAssetManager } from './nativeAssetManager';
+
+const AD_ANCHOR_CLASS_NAME = 'pb-click';
+const AD_DATA_ADID_ATTRIBUTE = 'pbAdId';
+
+export function newNativeRenderManager(win) {
+  let publisherDomain;
+
+
+  function findAdElements(className) {
+    let adElements = win.document.getElementsByClassName(className);
+    return adElements || [];
+  }
+
+  function loadClickTrackers(event, adId) {
+    fireTracker(adId, 'click');
+  }
+
+  function fireTracker(adId, action) {
+    if (adId === '') {
+      console.warn('Prebid tracking event was missing \'adId\'.  Was adId macro set in the HTML attribute ' + AD_DATA_ADID_ATTRIBUTE + 'on the ad\'s anchor element');
+    } else {
+      let message = { message: 'Prebid Native', adId: adId };
+
+      // fires click trackers when called via link
+      if (action === 'click') {
+        message.action = 'click';
+      }
+
+      win.parent.postMessage(JSON.stringify(message), publisherDomain);
+    }
+  }
+
+  function fireNativeImpTracker(adId){
+    fireTracker(adId, 'impression');
+  }
+
+  function fireNativeCallback(){
+    const adElements = findAdElements(AD_ANCHOR_CLASS_NAME);
+    for (let i = 0; i < adElements.length; i++) {
+      adElements[i].addEventListener('click', function(event) {
+        loadClickTrackers(event, window.pbNativeData.adId);
+      }, true);
+    }
+  }
+
+  // START OF MAIN CODE
+  let renderNativeAd = function(nativeTag){
+    window.pbNativeData = nativeTag;
+    const targetingData = transformAuctionTargetingData(nativeTag);
+    const nativeAssetManager = newNativeAssetManager(window);
+
+    if(nativeTag.hasOwnProperty('adId')) {
+      let parsedUrl = parseUrl(window.pbNativeData.pubUrl);
+      publisherDomain = parsedUrl.protocol + '://' + parsedUrl.host;
+
+      if(nativeTag.hasOwnProperty('rendererUrl') && !nativeTag.rendererUrl.match(/##.*##/i)){
+        const scr = document.createElement('SCRIPT');
+        scr.src = nativeTag.rendererUrl,
+        scr.id = 'pb-native-renderer';
+        document.body.appendChild(scr);
+      }
+      nativeAssetManager.loadAssets(nativeTag.adId,fireNativeCallback);
+      fireNativeCallback();
+      fireNativeImpTracker(nativeTag.adId);
+    }else{
```

Comment:
some more lint issues

Reply:
Updated

Reply:
Should have a space after closing bracket befor else and after

```javascript
} else {
```

Reply:
maybe just run enable eslint in your editor and see what else there is?